internal/core/metadata/application/notify.go

```diff
@@ -60,6 +60,24 @@ func addDeviceCallback(ctx context.Context, dic *di.Container, device dtos.Devic
 	go sendNotification(ctx, dic, device.Name, deviceCreateAction)
 }
 
+// validateDeviceCallback invoke device service's validation function for validating new device
+func validateDeviceCallback(ctx context.Context, dic *di.Container, device dtos.Device) errors.EdgeX {
+	lc := container.LoggingClientFrom(dic.Get)
+	deviceServiceCallbackClient, err := newDeviceServiceCallbackClient(ctx, dic, device.ServiceName)
+	if err != nil {
+		lc.Errorf("fail to create a device service callback client by serviceName %s, err: %v", device.ServiceName, err)
+		return err
+	}
+
+	request := requests.NewAddDeviceRequest(device)
```

Comment:
Please comment `way` using an `AddDeviceRequest` here.

Reply:
I'm a little confused would you elaborate more on this ?

Reply:
We are using `AddDeviceRequest` rather than creating a new `ValidateDeviceRequest` . Just need to comment why so those not aware of why are not confused.